examples/posenet/src/PoseNet.js

```diff
@@ -0,0 +1,372 @@
+    this._model.addOperand(scalarInt32Type);
+    this._model.setOperandValue(index, new Int32Array([value]));
+    return index;
+  }
+  _addScalarFloat32(value) {
+    const scalarInt32Type = {type: this._nn.FLOAT32};
+    let index = this._operandIndex++;
+    this._model.addOperand(scalarInt32Type);
+    this._model.setOperandValue(index, new Float32Array([value]));
+    return index;
+	}
+  
+  _addOpsAndParams(){
+    let index = 0;
+    for(let i in this._tfmodel){
+      if(this._tfmodel[i].convType=="conv2d"){
+        let inputs = [];
+        inputs.push(this._model._inputs[0]);
+        let outputs = this._outputs[index];
+        let paddingCode = 1;
+        inputs.push(outputs+1);
+        inputs.push(outputs+2);
+        inputs.push(this._addScalarInt32(paddingCode));
+        inputs.push(this._addScalarInt32(this._tfmodel[i].stride));
+        inputs.push(this._addScalarInt32(this._tfmodel[i].stride));
+        let fuseCode = 3; //relu6 activition function
+        inputs.push(this._addScalarInt32(fuseCode));
+        let opType = this._nn.CONV_2D;
+        this._model.addOperation(opType, inputs, [outputs]);
+      }		
+      else if(this._tfmodel[i].convType == "separableConv"){
+        //depthwise conv
+        let inputs = [];
+        inputs.push(this._outputs[index]);
+        index++;
+        let outputs = this._outputs[index];
+        let paddingCode = 1; //SAME
+        inputs.push(outputs+2);
+        inputs.push(outputs+3);
+        inputs.push(this._addScalarInt32(paddingCode));
+        inputs.push(this._addScalarInt32(this._tfmodel[i].stride));
+        inputs.push(this._addScalarInt32(this._tfmodel[i].stride));
+        inputs.push(this._addScalarInt32(1));
+        let fuseCode = 3;
+        inputs.push(this._addScalarInt32(fuseCode));
+        let opType = this._nn.DEPTHWISE_CONV_2D;
+        this._model.addOperation(opType, inputs, [outputs]);
+
+        //pointwise conv
+        let inputs_point = [];
+        inputs_point.push(this._outputs[index]);
+        index++;
+        let outputs_point = this._outputs[index];
+        paddingCode = 1;
+        inputs_point.push(outputs_point+3);
+        inputs_point.push(outputs_point+4);
+        inputs_point.push(this._addScalarInt32(paddingCode));
+        inputs_point.push(this._addScalarInt32(1));
+        inputs_point.push(this._addScalarInt32(1));
+        fuseCode = 3;
+        inputs_point.push(this._addScalarInt32(fuseCode));
+        opType = this._nn.CONV_2D;
+        this._model.addOperation(opType, inputs_point, [outputs_point]);
+      }
+    }	
+
+    if(this._type === "Multiperson"){
+      let paddingCode = 1;
+      let index = 5;
+      for(let i = 0; i<4; i++){
+        let inputs = [];
+        inputs.push(this._outputs[this._outputs.length-1]);
+        inputs.push(this._outputs[this._outputs.length-1]+index);
+        index++;
+        inputs.push(this._outputs[this._outputs.length-1]+index); 
+        index++;
```

Comment:
The above 5 lines of handling input tensor IDs look confusing. Could you please use meaningful data structure instead?